### PR TITLE
Handle tally cases where `results` is `None`

### LINKED ISSUE #1131: Results may be `None` during tallying
## Sentry link

https://openverse.sentry.io/share/issue/eaeee780feef4a96944d4e5ef4e6d39b/

## Description

It appears as though the value of `results` can be `None` during the tallying step:

```python
TypeError: 'NoneType' object is not subscriptable
  File "django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
  File "rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
  File "rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
  File "django/utils/decorators.py", line 46, in _wrapper
    return bound_method(*args, **kwargs)
  File "catalog/api/views/media_views.py", line 88, in list
    results, num_pages, num_results = search_controller.search(
  File "catalog/api/controllers/search_controller.py", line 429, in search
    results_to_tally = results[: 80 - (max_result_depth - page_size)]
```

This is right after a log line, `no image urls to validate`

## Reproduction

I _am_ able to reproduce this with the URL: https://api.openverse.engineering/v1/images/?license=by,by-nd,by-sa,cc0,pdm&license_type=commercial&filter_dead=true&mature=false&size=medium,large&page_size=100&page=1&q=kobinet

**Note**: You must use an authenticated app to request `page_size=100`; I tried pages 1-10 of the base `page_size` and I was unable to reproduce the error.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #1131 by @AetherUnbound

## Description

This PR initializes `results_to_tally` to either `results` or `[]` so that any operations on it which assume it's a list will behave correctly.

## Testing Instructions

1. Check out commit 65f2df5b8540eaae4ae2d7a40474620f3b5e5c72 and run `just api-test` - tests should fail and emulate current behavior
1. Check out the tip of this branch and run `just api-test` - tests should now pass!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by zackkrida — APPROVED** [review]
LGTM!

File: api/test/unit/controllers/test_search_controller.py [hunk 1]
```diff
@@ -463,3 +463,40 @@ def test_search_tallies_pages_less_than_5(
...
+    mock_post_process_results.return_value = None
+
+    serializer = MediaSearchRequestSerializer(data={"q": "dogs"})
+    serializer.is_valid()
+
+    search_controller.search(
+        search_params=serializer,
+        ip=0,
+        index=index,
+        page_size=100,  # High page size to trigger potential results past 80
```
Comment (sarayourfriend) [inline api/test/unit/controllers/test_search_controller.py:497]: Small change to clarify the situation:

```suggestion
        # Force calculated result depth length to include results within 80th position and above
        # to force edge case where retrieved results are only partially tallied.
        page=1,
        page_size=100,
```

**Review by sarayourfriend — APPROVED** [review]
LGTM. Just one suggestion for clarifying the test's scenario.
